packages/core-app-api/src/lib/AuthSessionManager/RefreshingAuthSessionManager.ts

```diff
@@ -83,10 +98,17 @@ export class RefreshingAuthSessionManager<T> implements SessionManager<T> {
         }
         return refreshedSession;
       } catch (error) {
-        if (options.optional) {
-          return undefined;
```

Comment:
This `optional` option means that we should never attempt to create a session, also not if the refresh fails